components/homme/CMakeLists.txt

```diff
@@ -314,10 +320,22 @@ IF (NOT (CPRNC_DIR))
 ENDIF ()
 
 IF (${HOMME_USE_KOKKOS})
+  INCLUDE(Kokkos)
   # Add unit tests for C++ code
   ADD_SUBDIRECTORY(test/unit_tests)
 ENDIF ()
 
+# Add libcompose, consisting of just C++ files, so we can build it just once for
+# all exes. It does not depend on compile-time sizes.
+if (HOMME_ENABLE_COMPOSE)
+  add_definitions(-DHOMME_ENABLE_COMPOSE)
+  set (COMPOSE_LIBRARY "compose")
+  add_subdirectory(src/share/compose)
+  set (USE_KOKKOS_KERNELS ON)
```

Comment:
Do we now use HOMME_USE_KOKKOS and USE_KOKKOS_KERNELS? Do not remember which one is for what.

Reply:
HOMMEXX uses USE_KOKKOS_KERNELS. I want to change that to HOMME_ENABLE_KOKKOS b/c KokkosKernels is itself a thing, so we shouldn't have a definition suggesting that KokkosKernels is being used. But that's a separate issue, one to discuss with the HOMMEXX team. In this PR, I just follow what HOMMEXX does to enable Kokkos in HOMME. In summary, HOMME_USE_KOKKOS and HOMME_ENABLE_KOKKOS are not definitions in master, nor are they introduced here.